File: go/isogram/isogram.go
package isogram

import (
	"strings"
	"unicode"
)

// IsIsogram determines if a word or phrase is an isogram.
func IsIsogram(word string) bool {
	var bitset uint32 = 0
	for _, c := range strings.ToLower(word) {
		if unicode.IsLetter(c) {
			if index := c - 'a'; bitset&(1<<index) == 0 {
				bitset |= 1 << index
			} else {
				return false
			}
		}
	}
	return true
}

File: c/isogram/src/isogram.c
#include "isogram.h"
#include <ctype.h>
#include <stdlib.h>

bool is_isogram(const char phrase[]) {
    if (phrase == NULL) return false;
    unsigned long set = 0;

    for (const char *c = phrase; *c != '\0'; ++c) {
        if (isalpha(*c)) {
            const char symbol = tolower(*c) - 'a';
            if (set & (1 << symbol)) return false;
            set |= 1 << symbol;
        }
    }
    return true;
}